### PR TITLE
Pipeline Controller and ISBService Controller reconciliation logic

### PR DESCRIPTION
**Pull request by juliev0**
This includes all of the reconciliation logic, with exception to the Status functions that Tony is working on (there are "todo" placeholders for those).

Tested by doing the following for both PipelineRollout and ISBServiceRollout:
- created Rollout object from our `samples` directory
- verified finalizer added to it
- verified ownerreference added to child object
- updated Rollout object spec (observed that child object changed, or in the case of ISBService, the numflow webhook denied the change as not allowed)
- deleted Rollout object and observed that child object got deleted

### CODE REVIEW

File: config/samples/numaplane.numaproj.io_v1alpha1_isbservicerollout.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 apiVersion: numaplane.numaproj.io/v1alpha1
 kind: ISBServiceRollout
 metadata:
-  name: isbservicerollout-sample
-  namespace: numaplane-system
+  name: my-isbsvc
+  namespace: example-namespace
```
Comment (juliev0) [inline config/samples/numaplane.numaproj.io_v1alpha1_isbservicerollout.yaml:5]: seems like a more realistic example to have a separate namespace, like one would be for an Asset

File: internal/controller/pipelinerollout_controller.go [hunk 2]
```diff
@@ -80,25 +85,94 @@ func (r *PipelineRolloutReconciler) Reconcile(ctx context.Context, req ctrl.Requ
...
-			Namespace: pipelineRollout.Namespace,
+			Name:            pipelineRollout.Name,
+			Namespace:       pipelineRollout.Namespace,
+			OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(pipelineRollout.GetObjectMeta(), apiv1.PipelineRolloutGroupVersionKind)},
 		},
 		Spec: pipelineRollout.Spec.Pipeline,
 	}
 
-	err := kubernetes.UpdateCRSpec(ctx, r.restConfig, &obj, "pipelines")
+	err := kubernetes.ApplyCRSpec(ctx, r.restConfig, &obj, "pipelines")
```
Comment (juliev0) [inline internal/controller/pipelinerollout_controller.go:154]: just a name change

File: config/samples/numaplane.numaproj.io_v1alpha1_pipelinerollout.yaml [hunk 3]
```diff
@@ -1,9 +1,10 @@
 apiVersion: numaplane.numaproj.io/v1alpha1
 kind: PipelineRollout
 metadata:
-  name: pipelinerollout-example
-  namespace: numaplane-system
+  name: my-pipeline
+  namespace: example-namespace
 spec:
+  interStepBufferServiceName: my-isbsvc
```
Comment (afugazzotto) [inline config/samples/numaplane.numaproj.io_v1alpha1_pipelinerollout.yaml:7]: This needs to be added to the pipelinerollout_types.go file in the PipelineRolloutSpec struct.

Comment (juliev0) [reply config/samples/numaplane.numaproj.io_v1alpha1_pipelinerollout.yaml:7]: oh, sorry, this is a typo: I meant to put it underneath `pipeline` below

Comment (juliev0) [reply config/samples/numaplane.numaproj.io_v1alpha1_pipelinerollout.yaml:7]: good catch

**Review by afugazzotto — APPROVED** [review]

**Review by xdevxy — APPROVED** [review]

**Pull request merged**